### PR TITLE
Support underscores (in addition to hyphens) for charset detection.

### PR DESCRIPTION
**Pull request by srividyut**
This is the issue of [#10307](https://github.com/matrix-org/synapse/issues/10307).
This is the resurrection of closed pull request #10306.
(I "dirty the branch myself" & "too long" so I decided to start over.) 
This is a slight fix to the character code detection of the URL-Preview function.
See [this issue #10307](https://github.com/matrix-org/synapse/issues/10307) for details. 

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#changelog). The entry should:
* [x] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#sign-off)
* [x] Code style is correct (run the [linters](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#code-style))
I run "./scripts-dev/lint.sh" and it looks like this: 
  All done! ✨ 🍰 ✨

### CODE REVIEW

File: changelog.d/10410.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+Slight improvements in character set detection for the "URL_Preview" feature.
```
Comment (clokep) [inline changelog.d/10410.bugfix:1]:
```suggestion
Improve character set detection in URL previews by supporting underscores (in addition to hyphens). Contributed by @srividyut.
```

**Review by clokep — COMMENTED** [review]
Looks good overall! Could you try to add an additional test to `tests.test_preview.MediaEncodingTestCase` which ensures that both underscores and hyphens work?

File: changelog.d/10410.bugfix (hunk 1 above)
Comment (srividyut) [reply changelog.d/10410.bugfix:1]: Ah, I didn't have enough "work to run trial tests.test_preview.MediaEncodingTestCase".
I will definitely try it.

**Comment by srividyut** [comment]
Ah, I didn't have enough "work to run trial tests.test_preview.MediaEncodingTestCase".
I will definitely try it.

File: tests/test_preview.py [hunk 2]
```diff
@@ -325,6 +325,18 @@ def test_meta_charset(self):
         )
         self.assertEqual(encoding, "ascii")
 
+        """A character encoding contains underscore."""
```
Comment (clokep) [inline tests/test_preview.py:328]:
```suggestion
    def test_meta_charset_underscores(self):
        """A character encoding contains underscore."""
```

**Review by clokep — APPROVED** [review]
LGTM if tests pass!

**Comment by clokep** [comment]
Thank you so much! 🎉

**Pull request merged**